---
title: "Özel Durumlar: failwith İşlevi (F#)"
description: "Nasıl 'failwith' işlevi bir F # özel durum oluşturur öğrenin."
keywords: "Visual f #, f # işlevsel programlama"
author: cartermp
ms.author: phcart
ms.date: 05/16/2016
ms.topic: language-reference
ms.prod: .net
ms.technology: devlang-fsharp
ms.devlang: fsharp
ms.assetid: 2e0c1f28-cc6c-4ecd-bb93-3816c4dd7cd3
ms.openlocfilehash: 295a4d754e0df015ba67daa9cf80d7df5b40199c
ms.sourcegitcommit: ffb9aa26cd5211dc6d96ebb42968d8357048880e
ms.translationtype: MT
ms.contentlocale: tr-TR
ms.lasthandoff: 11/09/2017
---
# <a name="exceptions-the-failwith-function"></a><span data-ttu-id="da924-104">Özel Durumlar: failwith İşlevi</span><span class="sxs-lookup"><span data-stu-id="da924-104">Exceptions: The failwith Function</span></span>

<span data-ttu-id="da924-105">`failwith` İşlevi F # özel durum oluşturur.</span><span class="sxs-lookup"><span data-stu-id="da924-105">The `failwith` function generates an F# exception.</span></span>


## <a name="syntax"></a><span data-ttu-id="da924-106">Sözdizimi</span><span class="sxs-lookup"><span data-stu-id="da924-106">Syntax</span></span>

```fsharp
failwith error-message-string
```

## <a name="remarks"></a><span data-ttu-id="da924-107">Açıklamalar</span><span class="sxs-lookup"><span data-stu-id="da924-107">Remarks</span></span>
<span data-ttu-id="da924-108">*Hata ileti dizesi* önceki sözdiziminde sabit değerli bir dize türünde bir değer mi `string`.</span><span class="sxs-lookup"><span data-stu-id="da924-108">The *error-message-string* in the previous syntax is a literal string or a value of type `string`.</span></span> <span data-ttu-id="da924-109">Bu duruma `Message` özel durum özelliği.</span><span class="sxs-lookup"><span data-stu-id="da924-109">It becomes the `Message` property of the exception.</span></span>

<span data-ttu-id="da924-110">Tarafından oluşturulan özel durum `failwith` olan bir `System.Exception` adına sahip bir başvuru özel durumu `Failure` F # kodunda.</span><span class="sxs-lookup"><span data-stu-id="da924-110">The exception that is generated by `failwith` is a `System.Exception` exception, which is a reference that has the name `Failure` in F# code.</span></span> <span data-ttu-id="da924-111">Aşağıdaki kod kullanımını göstermektedir `failwith` bir özel durum için.</span><span class="sxs-lookup"><span data-stu-id="da924-111">The following code illustrates the use of `failwith` to throw an exception.</span></span>

[!code-fsharp[Main](../../../../samples/snippets/fsharp/lang-ref-2/snippet6001.fs)]
    
## <a name="see-also"></a><span data-ttu-id="da924-112">Ayrıca Bkz.</span><span class="sxs-lookup"><span data-stu-id="da924-112">See Also</span></span>
[<span data-ttu-id="da924-113">Özel durum işleme</span><span class="sxs-lookup"><span data-stu-id="da924-113">Exception Handling</span></span>](index.md)

[<span data-ttu-id="da924-114">Özel durum türleri</span><span class="sxs-lookup"><span data-stu-id="da924-114">Exception Types</span></span>](exception-types.md)

[<span data-ttu-id="da924-115">Özel durumlar: `try...with` ifade</span><span class="sxs-lookup"><span data-stu-id="da924-115">Exceptions: The `try...with` Expression</span></span>](the-try-with-expression.md)

[<span data-ttu-id="da924-116">Özel durumlar: `try...finally` ifade</span><span class="sxs-lookup"><span data-stu-id="da924-116">Exceptions: The `try...finally` Expression</span></span>](the-try-finally-expression.md)

[<span data-ttu-id="da924-117">Özel durumlar: `raise` işlevi</span><span class="sxs-lookup"><span data-stu-id="da924-117">Exceptions: the `raise` Function</span></span>](the-raise-function.md)
